Display guest full name in "html" plugin report

happz opened the pull request:
To visualy connect results with the guest - with multi-host in queue, a single test may appear multiple times, this should be a nice visual aid.

lukaszachy reviewed (changes requested):
Good idea, however could we detect if that column is necessary (== there are multiple guests used)?

The repeated 'default-0' value in the whole column just steals space.

![image](https://user-images.githubusercontent.com/1548503/225258921-7e69fad2-9931-42b7-955b-9b23cf76b993.png)

happz commented:
> Good idea, however could we detect if that column is necessary (== there are multiple guests used)?
> 
> The repeated 'default-0' value in the whole column just steals space.
> 
> ![image](https://user-images.githubusercontent.com/1548503/225258921-7e69fad2-9931-42b7-955b-9b23cf76b993.png)

I suppose the same objection can be raised against https://github.com/teemtee/tmt/pull/1909.

I'm not sure what the default behavior should be, my natural tendency against exceptions would prefer keeping the column around, even if it's uniform like this, but I don't mind either variant. Although I'd add an option to control it, whatever the default we end up with.

kkaarreell commented:
Can we have something like this also in the junit?

kkaarreell commented:
> The repeated 'default-0' value in the whole column just steals space.

IMHO there is still plenty of free space in that report. ;-)

lukaszachy commented:
> I'm not sure what the default behavior should be, my natural tendency against exceptions would prefer keeping the column around, even if it's uniform like this, but I don't mind either variant. Although I'd add an option to control it, whatever the default we end up with.

Option sounds good. I do not mind having it enabled by default as long as I can disable this :D

happz commented:
Rebased on top of https://github.com/teemtee/tmt/pull/1903, and added the option and a test:

```
  --display-guest always|multihost|never
                                  When to display full guest name in report:
                                  always, when more than a single guest was
                                  involved, or never.
```

I'm not very happy about the "multihost" string - I wanted to convey the "display guest name only when there's more than one guest involved in the plan", but I'm pretty sure "multihost" is not the final answer :/

happz commented:
> Can we have something like this also in the junit?

Filed https://github.com/teemtee/tmt/issues/1924 - I can implement it, but I don't know how it should be done, so any hints are welcome.

lukaszachy commented:
> I'm not very happy about the "multihost" string - I wanted to convey the "display guest name only when there's more than one guest involved in the plan", but I'm pretty sure "multihost" is not the final answer :/

How about 'auto' ? (used almost as often as 'data':)

happz commented:
Ok, `auto` it is.

thrix commented on tmt/steps/report/html.py:
```diff
@@ -31,6 +31,15 @@ class ReportHtmlData(tmt.steps.report.ReportStepData):
         )
 
+    display_guest: str = field(
+        default='always',
```
@happz isn't `auto` a better default option? I would opt for it

happz replied:
In my books, `always` is much better because it's always the same output, no matter how many guests are involved. Let's wait for someone to decide, @psss @lukaszachy care to chime in?

lukaszachy replied:
I vote for 'always' to have stable column names

psss replied:
I'd vote for `auto` as @thrix suggested. Let's make the default view concise if just a single guest is involved.

happz replied:
Changed in e69ea8e2

lukaszachy replied:
I respect the voting ;)

happz replied:
I believe this can be now closed.

thrix reviewed (changes requested):
I would propose to use `auto` as default

lukaszachy reviewed (changes requested):
'schema' for html report should be updated too, right?

happz commented:
> 'schema' for html report should be updated too, right?

Absolutely, thanks for spotting this!

happz commented:
/packit build

happz commented:
/packit retest-failed

The-Mule commented:
I am bit of lost  - is this really just about guest hostname or what? I am afraid see hostname won't help much for mutlihost testing use case. I suppose in case of multihost testing we would benefit more from seeing a role there (perhaps as a guest hostname prefix / suffix). Or did I misunderstood what "guest full name" is?

happz commented:
> I am bit of lost - is this really just about guest hostname or what? I am afraid see hostname won't help much for mutlihost testing use case. I suppose in case of multihost testing we would benefit more from seeing a role there (perhaps as a guest hostname prefix / suffix). Or did I misunderstood what "guest full name" is?

It's definitely not about the hostname :) The patch adds the guest *name* (i.e. `name: ...` from the `provision` step of your plan, or `default-N` if you didn't specify) and guest role, if any (i.e. `role: ...` from the same place) into the HTML report. It's called "full name", but that was a mistake and with https://github.com/teemtee/tmt/pull/1943 we would call these two bits a guest's "multihost name".

thrix reviewed (approved):
LGTM

psss reviewed (approved):
Nice improvement, thanks!
